### ISSUE TITLE
Automatic matching for local macros

### ISSUE THREAD

**Issue by gaksaray** [hugetim/nbstata#20]
`nbstata` on Jupyter Notebook, is it possible to have a language definition for local macros such that closing ticks are automatically inserted (as in Stata do-file editor). Currently, there is autocompletion for double and single quotes, but not Stata macros:

https://user-images.githubusercontent.com/72267919/229855770-bc63a35e-d0c5-4b0d-aae6-f6e587b2b789.mov

What's worse is when the user inserts closing macro tick, Jupyter automatically adds another spurious single quote, which must be deleted manually.

**Comment by hugetim** [comment]
A. I think you should be able to disable quote matching in your Jupyter config, as a workaround for the worse part.

B. Configuring local macro open/close matching may be possible via an extension that tweaks CodeMirror (the code editor plugin used by Jupyter), but that would take some work to figure out.

C. Autocomplete in Jupyter is something a little different, properly speaking. That behavior, tab-autocompletion, *is* determined directly by the `nbstata` kernel code. And it already should provide the closing single-quote for you, as demonstrated by the test shown [here](https://github.com/hugetim/nbstata/blob/master/nbs/11_completions.ipynb):
```python
completions_test_setup('local test_local "test value"')
test_eq(
    _complete("list `t", 7),
    ["test_local'"],
)
```
In other words, if you define such a local macro in a previous code cell and then, in a new code cell, start typing the following
```stata
list `t
```
and then hit the Tab key, it should fill in `est_local'` for you (or offer a dropdown if there are multiple possible completions).

**Comment by gaksaray** [comment]
I tried C (autocompletion of macros after first backtick). It works!

I couldn't figure out A.

About B (automatic local macro open/close matching). This is already implemented in [jupyterlab_stata_highlight2](https://github.com/hugetim/jupyterlab_stata_highlight2) for Jupyter Lab. Maybe there is an easy way to extend the same behavior to Notebook interface?

**Comment by hugetim** [comment]
Related to #17
